web_src/js/features/repo-issue.js

```diff
@@ -673,19 +673,28 @@ export function initRepoIssueBranchSelect() {
   });
 }
 
-export function initSingleCommentEditor($commentForm) {
+export async function initSingleCommentEditor($commentForm) {
   // pages:
   // * normal new issue/pr page, no status-button
   // * issue/pr view page, with comment form, has status-button
   const opts = {};
   const statusButton = document.querySelector('#status-button');
-  if (statusButton) {
+  const commentButton = document.querySelector('#comment-button');
+  if (statusButton || commentButton) {
     opts.onContentChanged = (editor) => {
-      const statusText = statusButton.getAttribute(editor.value().trim() ? 'data-status-and-comment' : 'data-status');
-      statusButton.textContent = statusText;
+      const editorText = editor.value().trim();
+      if (statusButton) {
+        const statusText = statusButton.getAttribute(editorText ? 'data-status-and-comment' : 'data-status');
+        statusButton.textContent = statusText;
+      }
+      if (commentButton) {
+        commentButton.disabled = !editorText;
+      }
     };
   }
-  initComboMarkdownEditor($commentForm.find('.combo-markdown-editor'), opts);
+  const editor = await initComboMarkdownEditor($commentForm.find('.combo-markdown-editor'), opts);
+  // initialize comment button
+  opts.onContentChanged(editor);
```

Comment:
It might throw errors. I will make some improvements.

Reply:
Please help to take a look whether cdf2798680bdf74016231a2747a420b5b069171f looks good to you

Reply:
This looks good; also manually tested the new/view issue/PR forms just to be sure and they work fine.

I did throw most of the logic into that `statusButton || commentButton` if statement because for new issue/PR forms the if statements in `onContentChanged()` would never be true. I'm fine with your approach because that optimization is negligible and this reads alot better

Reply:
The major difference is:

old: 

```
const opts = {};
if (...) opts.onContentChanged = ...
opts.onContentChanged(...)
```

The problem is, the `opts.onContentChanged()` call might fail because `onContentChanged` isn't set.

So I changed the logic.

Reply:
Nice catch, not sure why I didn't didn't see that